gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/web/WebExtension.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020-2021 JetBrains s.r.o. and respective authors and developers.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package org.jetbrains.compose.web
+
+import org.gradle.api.Project
+import org.gradle.api.plugins.ExtensionAware
+import org.jetbrains.compose.internal.mppExt
+import org.jetbrains.kotlin.gradle.plugin.KotlinTarget
+import org.jetbrains.kotlin.gradle.targets.js.ir.KotlinJsIrTarget
+
+abstract class WebExtension : ExtensionAware {
+    private var requestedTargets: Set<KotlinJsIrTarget>? = null
+    private var targetsToConfigure: Set<KotlinJsIrTarget>? = null
+
+    // public api
+    @Suppress("unused")
+    fun targets(vararg targets: KotlinTarget) {
+        check(requestedTargets == null) {
+            "compose.web.targets() was already set!"
+        }
+
+        val jsIrTargets = linkedSetOf<KotlinJsIrTarget>()
+        for (target in targets) {
+            check(target is KotlinJsIrTarget) {
+                """|'${target.name}' is not a JS(IR) target:
+                |* add `kotlin.js.compiler=ir` to gradle properties;
+                |* define target as `kotlin { js(IR) { ... } }`
+            """.trimMargin()
+            }
+            jsIrTargets.add(target)
+        }
+        requestedTargets = jsIrTargets
+    }
+
+    internal fun targetsToConfigure(project: Project): Set<KotlinJsIrTarget> {
+        targetsToConfigure =
+            targetsToConfigure
+                ?: requestedTargets
+                ?: defaultJsTargetsToConfigure(project)
+
+        return targetsToConfigure!!
+    }
+
+    private fun defaultJsTargetsToConfigure(project: Project): Set<KotlinJsIrTarget> {
+        val mppTargets = project.mppExt?.targets?.asMap?.values ?: emptySet()
+        val jsIRTargets: Set<KotlinJsIrTarget> = mppTargets.filterIsInstanceTo(LinkedHashSet())
```

Comment:
Maybe pass type argument to the right?
`val jsIRTargets = mppTargets.filterIsInstanceTo(mutableSetOf<KotlinJsIrTarget>())`

Reply:
OK